code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -5112,6 +5112,217 @@
+	M.hallucination = max(M.hallucination, 10)
+	M.halloss += 5
+
+/datum/reagent/ethanol/electric_sheep
+	name = "Electric Sheep"
+	id = ELECTRIC_SHEEP
+	description = "This is what robots dream about."
+	color = "#664300"
+	custom_metabolism = 1
+
+/datum/reagent/ethanol/electric_sheep/on_mob_life(var/mob/living/M) //If it's human, shoot sparks every tick! If MoMMI, cause alcohol effects.
+	if(..())
+		return 1
+	if(ishuman(M))
+		spark(M, 5, FALSE)
+
+/datum/reagent/ethanol/electric_sheep/reaction_mob(var/mob/living/M)
+	if(isrobot(M))
+		M.Jitter(20)
+		M.Dizzy(20)
+		M.druggy = max(M.druggy, 60)
+
+/datum/reagent/ethanol/suicide
+	name = "Suicide"
+	id = SUICIDE
+	description = "It's only tolerable because of the added alcohol."
+	color = "#664300"
+	custom_metabolism = 2
+
+/datum/reagent/ethanol/suicide/on_mob_life(var/mob/living/M)  //Instant vomit. Every tick.
+	if(..())
+		return 1
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.vomit(0,1)
+
+/datum/reagent/ethanol/metabuddy
+	name = "Metabuddy"
+	id = METABUDDY
+	description = "Ban when?"
+	color = "#664300"
+	var/global/list/datum/mind/metaclub = list()
+
+/datum/reagent/ethanol/metabuddy/on_mob_life(var/mob/living/L)
+	if(..())
+		return 1
+	var/datum/mind/LM = L.mind
+	if(!metaclub.Find(LM) && LM)
+		metaclub += LM
+		var/datum/mind/new_buddy = LM
+		for(var/datum/mind/M in metaclub) //Update metaclub icons
+			if(M.current.client && new_buddy.current && new_buddy.current.client)
+				var/imageloc = new_buddy.current
+				var/imagelocB = M.current
+				if(istype(M.current.loc,/obj/mecha))
+					imageloc = M.current.loc
+					imagelocB = M.current.loc
+				var/image/I = image('icons/mob/mob.dmi', loc = imageloc, icon_state = "metaclub")
+				I.plane = REV_ANTAG_HUD_PLANE
+				M.current.client.images += I
+				var/image/J = image('icons/mob/mob.dmi', loc = imagelocB, icon_state = "metaclub")
+				J.plane = REV_ANTAG_HUD_PLANE
+				new_buddy.current.client.images += J
+
+/datum/reagent/ethanol/waifu
+	name = "Waifu"
+	id = WAIFU
+	description = "Don't drink more than one waifu if you value your laifu."
+	color = "#664300"
+
+/datum/reagent/ethanol/waifu/on_mob_life(var/mob/living/M)
+	if(..())
+		return 1
+	if(M.gender == MALE)
+		M.setGender(FEMALE)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(!M.is_wearing_item(/obj/item/clothing/under/schoolgirl))
+			var/turf/T = get_turf(H)
+			T.turf_animation('icons/effects/96x96.dmi',"beamin",-32,0,MOB_LAYER+1,'sound/effects/rejuvinate.ogg',anim_plane = MOB_PLANE)
+			H.visible_message("<span class='warning'>[H] dons her magical girl outfit in a burst of light!</span>")
+			var/obj/item/clothing/under/schoolgirl/S = new /obj/item/clothing/under/schoolgirl(get_turf(H))
+			if(!H.put_in_hands(S) && H.w_uniform) //For quickswap, it needs to be in a hand for some reason. But...
+				H.w_uniform.forceMove(get_turf(H)) //If both hands are occupied, we have another solution: just drop the current suit (empties ID and pockets)
```

Comment:
Use H.u_equip(H.w_uniform, 1) instead, so you can remove the w_uniform.unequipped below